components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeService.java

```diff
@@ -306,6 +319,292 @@ public Scope updateScope(Scope updatedScope) throws IdentityOAuth2ScopeException
+     *
+     * @param userId    User Id.
+     * @param appId     Application Id.
+     * @param tenantId  Tenant Id.
+     * @throws IdentityOAuth2ScopeConsentException
+     */
+    public void revokeUserConsentForApplication(String userId, String appId, int tenantId)
+            throws IdentityOAuth2ScopeConsentException {
+
+        validateUserId(userId);
+        validateAppId(appId);
+        try {
+            OAuthTokenPersistenceFactory.getInstance().getOAuthUserConsentedScopesDAO()
+                    .deleteUserConsentOfApplication(userId, appId, tenantId);
+            if (log.isDebugEnabled()) {
+                log.debug("Successfully revoked the user consents for OAuth scopes for user : " + userId +
+                        " and application : " + appId + " for tenant with Id : " + tenantId);
+            }
+        } catch (IdentityOAuth2ScopeConsentException e) {
+            String msg = String.format(ERROR_CODE_FAILED_TO_REVOKE_USER_CONSENT_FOR_APP.getMessage(), userId, appId,
+                    tenantId);
+            throw new IdentityOAuth2ScopeConsentException(
+                    ERROR_CODE_FAILED_TO_REVOKE_USER_CONSENT_FOR_APP.getCode(), msg, e);
+        }
+    }
+
+    /**
+     * Revoke all scope consents for the user.
+     *
+     * @param userId    User Id.
+     * @param tenantId  Tenant Id.
+     * @throws IdentityOAuth2ScopeConsentException
+     */
+    public void revokeUserConsents(String userId, int tenantId)
+            throws IdentityOAuth2ScopeConsentException {
+
+        validateUserId(userId);
+        try {
+            OAuthTokenPersistenceFactory.getInstance().getOAuthUserConsentedScopesDAO()
+                    .deleteUserConsents(userId, tenantId);
+            if (log.isDebugEnabled()) {
+                log.debug("Successfully deleted the user consents OAuth scopes for user : " + userId +
+                        " in tenant with Id : " + tenantId);
+            }
+        } catch (IdentityOAuth2ScopeConsentException e) {
+            String msg = String.format(ERROR_CODE_FAILED_TO_REVOKE_USER_CONSENT.getMessage(), userId, tenantId);
+            throw new IdentityOAuth2ScopeConsentException(
+                    ERROR_CODE_FAILED_TO_REVOKE_USER_CONSENT.getCode(), msg, e);
+        }
+    }
+
+    /**
+     * Check if user has already consented for requested scopes.
+     *
+     * @param userId                            User Id.
+     * @param appId                             Application Id.
+     * @param tenantId                          Tenant Id.
+     * @param consentRequiredApprovedScopes     List of consent required approved scopes.
+     * @param consentRequiredDeniedScopes       List of consent required denied scopes.
+     * @return true if user has already provided the consent.
+     * @throws IdentityOAuth2ScopeConsentException
+     */
+    public boolean hasUserAlreadyProvidedConsentForAllRequestedScopes(String userId, String appId,
```

Comment:
Why do we need the word already here ? Doesn't hasUserProvidedConsentForAllRequestedScopes do the work ?